lib/bolt/config.rb

```diff
@@ -183,6 +184,18 @@ def initialize(project, config_data, overrides = {})
 
       override_data = normalize_overrides(overrides)
 
+      # If we need to lower concurrency and concurrency is not configured
+      ld_concurrency = loaded_data.map(&:keys).flatten.include?('concurrency')
+      if default_concurrency != DEFAULT_DEFAULT_CONCURRENCY &&
+         !ld_concurrency &&
+         !override_data.key?('concurrency')
+        concurrency_warning = { option: 'concurrency',
+                                msg: "Concurrency will default to #{default_concurrency} because ulimit "\
+                                "is low: #{Etc.sysconf(Etc::SC_OPEN_MAX)}. Set concurrency with "\
+                                "'--concurrency', or set your ulimit with 'ulimit -n <limit>'" }
```

Comment:
I wonder if we could send this to `info` instead of `warn` and update the docs to say concurrency goes _up to_ 100.. a warning for a default value seems a little weird to me, especially since we'll have a similar message to this when a low ulimit actually becomes a problem that the user needs to make adjustments for.

Reply:
I think it's appropriate to warn here because it can cause unexpected behavior, particularly if a user typically runs bolt on a machine that has a higher ulimit - namely setting the concurrency lower can cause Bolt to run more slowly. I think of info messages as being useful but not necessary, things like what steps are being performed or where modules are being loaded from. This message feels slightly more important for users to be aware of and potentially take action on. It's not an error because nothing is wrong, but it's probably not a desirable state for Bolt to run in.

I do think I can update the default in the docs to be "100 or 1/3 of ulimit, whichever is lower"